packages/nuxt/src/core/plugins/resolve-deep-imports.ts

```diff
@@ -29,12 +32,27 @@ export function resolveDeepImportsPlugin (nuxt: Nuxt): Plugin {
         resolvedConditions.add('require')
       }
       conditions = [...resolvedConditions]
+
+      const { runtimeDependencies = [] } = await tryImportModule<typeof import('nitro/runtime/meta')>('nitro/runtime/meta', {
+        url: nuxt.options.modulesDir.map(d => directoryToURL(d)),
```

Comment:
(nitro is explicit dep, it avoid hoisting issues with nitro pkg itself)

```suggestion
        url: import.meta.url,
```